pkg/agent/cniserver/server.go

```diff
@@ -384,13 +391,46 @@ func (s *CNIServer) GetPodConfigurator() *podConfigurator {
 	return s.podConfigurator
 }
 
+// Antrea IPAM for secondary network.
+func (s *CNIServer) ipamAdd(cniConfig *CNIConfig) (*cnipb.CniCmdResponse, error) {
+	ipamResult, err := ipam.SecondaryNetworkAdd(cniConfig.CniCmdArgs, cniConfig.K8sArgs, cniConfig.NetworkConfig)
+	if err != nil {
+		return s.ipamFailureResponse(err), nil
+	}
+	klog.InfoS("Allocated IP addresses", "container", cniConfig.ContainerId, "result", ipamResult)
+	return resultToResponse(ipamResult), nil
+}
+
+func (s *CNIServer) ipamDel(cniConfig *CNIConfig) (*cnipb.CniCmdResponse, error) {
+	if err := ipam.SecondaryNetworkDel(cniConfig.CniCmdArgs, cniConfig.K8sArgs, cniConfig.NetworkConfig); err != nil {
+		return s.ipamFailureResponse(err), nil
+	}
+	return &cnipb.CniCmdResponse{CniResult: []byte("")}, nil
+}
+
+func (s *CNIServer) ipamCheck(cniConfig *CNIConfig) (*cnipb.CniCmdResponse, error) {
+	if err := ipam.SecondaryNetworkCheck(cniConfig.CniCmdArgs, cniConfig.K8sArgs, cniConfig.NetworkConfig); err != nil {
```

Comment:
Since ipam.SecondaryNetworkCheck always returns error, line 415 won't execute.

Reply:
Yes, but I hope to keep the ipamCheck() complete and independent of the SecondaryNetworkCheck() implementation. I added some comments to explain this.